addons/stock_account/report/stock_history.py

```diff
@@ -58,9 +58,13 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
                 stock_histories_by_group[key] += [line['id']]
 
             histories_dict = {}
-            not_real_cost_method_products = self.env['product.product'].browse(
-                record['product_id'] for record in stock_history_data.values()
-            ).filtered(lambda product: product.cost_method != 'real')
+            product_no_prefetch = self.env['product.product'].with_context(prefetch_fields=False)
+            product_ids = set([])
+            stock_history_product_ids = (record['product_id'] for record in stock_history_data.values())
+            for product in product_no_prefetch.browse(stock_history_product_ids):
+                if product.cost_method != 'real':
```

Comment:
would using product_no_prefetch.read(['cost_method']) gain something?

Reply:
@gurneyalex do you think it makes a difference because it will only get one column instead of the greedy default column loading from Odoo?

Reply:
actually with prefetch_fields=False in the context, it will not improve things.